File: chat_logger.js
const fs = require('fs');
const path = require('path');

class ChatLogger {
    constructor(baseDir = 'chat-history') {
        this.baseDir = baseDir;
        this.logsDir = path.join(baseDir, 'logs');
        this.summariesDir = path.join(baseDir, 'summaries');
        
        if (!fs.existsSync(this.logsDir)) {
            fs.mkdirSync(this.logsDir, { recursive: true });
        }
        if (!fs.existsSync(this.summariesDir)) {
            fs.mkdirSync(this.summariesDir, { recursive: true });
        }
    }

    logSessionStart() {
        const now = new Date();
        const sessionId = now.toISOString().replace(/[:.]/g, '-').slice(0, 19);
        this.currentSessionId = sessionId;
        
        const logEntry = {
            session_id: sessionId,
            start_time: now.toISOString(),
            status: 'started'
        };
        
        const logFile = path.join(this.logsDir, `${sessionId}.json`);
        fs.writeFileSync(logFile, JSON.stringify(logEntry, null, 2), 'utf-8');
        
        return sessionId;
    }

    logModification(filePath, description, changes) {
        if (!this.currentSessionId) {
            this.logSessionStart();
        }
        
        const timestamp = new Date().toISOString();
        const logFile = path.join(this.logsDir, `${this.currentSessionId}.json`);
        
        let data = {};
        try {
            const content = fs.readFileSync(logFile, 'utf-8');
            data = JSON.parse(content);
        } catch (e) {
            data = { session_id: this.currentSessionId, modifications: [] };
        }
        
        if (!data.modifications) {
            data.modifications = [];
        }
        
        data.modifications.push({
            timestamp: timestamp,
            file: filePath,
            description: description,
            changes: changes
        });
        
        fs.writeFileSync(logFile, JSON.stringify(data, null, 2), 'utf-8');
    }

    saveSummary(summaryText) {
        if (!this.currentSessionId) {
            this.logSessionStart();
        }
        
        const summaryFile = path.join(this.summariesDir, `${this.currentSessionId}_summary.md`);
        const now = new Date();
        const content = `# Resumen de Sesión: ${this.currentSessionId}\n\n**Fecha:** ${now.toLocaleString()}\n\n${summaryText}`;
        
        fs.writeFileSync(summaryFile, content, 'utf-8');
    }
}

// Prueba del sistema
const logger = new ChatLogger();
logger.logSessionStart();
logger.logModification('chat_logger.js', 'Creación del logger en JavaScript', ['Implementación de clase', 'Métodos de registro']);
logger.saveSummary('Se ha implementado y probado el sistema de logging para el historial de chats.');
console.log('Sistema de logging probado exitosamente.');

File: chat_logger.py
import os
import json
from datetime import datetime

class ChatLogger:
    def __init__(self, base_dir="chat-history"):
        self.base_dir = base_dir
        self.logs_dir = os.path.join(base_dir, "logs")
        self.summaries_dir = os.path.join(base_dir, "summaries")
        os.makedirs(self.logs_dir, exist_ok=True)
        os.makedirs(self.summaries_dir, exist_ok=True)

    def log_session_start(self):
        timestamp = datetime.now()
        session_id = timestamp.strftime("%Y%m%d_%H%M%S")
        self.current_session_id = session_id
        
        log_entry = {
            "session_id": session_id,
            "start_time": timestamp.isoformat(),
            "status": "started"
        }
        
        log_file = os.path.join(self.logs_dir, f"{session_id}.json")
        with open(log_file, 'w', encoding='utf-8') as f:
            json.dump(log_entry, f, indent=2, ensure_ascii=False)
            
        return session_id

    def log_modification(self, file_path, description, changes):
        if not hasattr(self, 'current_session_id'):
            self.log_session_start()
            
        timestamp = datetime.now().isoformat()
        log_file = os.path.join(self.logs_dir, f"{self.current_session_id}.json")
        
        try:
            with open(log_file, 'r', encoding='utf-8') as f:
                data = json.load(f)
        except (FileNotFoundError, json.JSONDecodeError):
            data = {"session_id": self.current_session_id, "modifications": []}
            
        if "modifications" not in data:
            data["modifications"] = []
            
        data["modifications"].append({
            "timestamp": timestamp,
            "file": file_path,
            "description": description,
            "changes": changes
        })
        
        with open(log_file, 'w', encoding='utf-8') as f:
            json.dump(data, f, indent=2, ensure_ascii=False)

    def save_summary(self, summary_text):
        if not hasattr(self, 'current_session_id'):
            self.log_session_start()
            
        summary_file = os.path.join(self.summaries_dir, f"{self.current_session_id}_summary.md")
        with open(summary_file, 'w', encoding='utf-8') as f:
            f.write(f"# Resumen de Sesión: {self.current_session_id}\n\n")
            f.write(f"**Fecha:** {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n\n")
            f.write(summary_text)

# Instancia global para uso fácil
logger = ChatLogger()
